[CI] KibanaUserRoleIntegTests testFieldMappings failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #4171 / ubuntu-2004-aarch64_checkpart4_platform-support-arm](https://gradle-enterprise.elastic.co/s/ibrbsw52aopl4)
- [elasticsearch-periodic-platform-support #4115 / oraclelinux-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/4gmewo7a5sbiq)
- [elasticsearch-pull-request #32171 / part-4](https://gradle-enterprise.elastic.co/s/2rnsewyv6bkgg)
- [elasticsearch-pull-request #32011 / part-4](https://gradle-enterprise.elastic.co/s/slykfu5uw7d7k)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:security:internalClusterTest" --tests "org.elasticsearch.integration.KibanaUserRoleIntegTests.testFieldMappings" -Dtests.seed=44DBD42710118171 -Dtests.locale=rm-CH -Dtests.timezone=Antarctica/Troll -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testFieldMappings'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.integration.KibanaUserRoleIntegTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: safeAwait: CountDownLatch did not reach zero within the timeout
```

**Issue Reasons:**
- [main] 4 failures in test testFieldMappings (0.7% fail rate in 596 executions)
- [main] 2 failures in step part-4 (0.9% fail rate in 220 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-platform-support (18.2% fail rate in 11 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (0.9% fail rate in 216 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -52,12 +52,6 @@ tests:
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=mtermvectors/10_basic/Tests catching other exceptions per item}
   issue: https://github.com/elastic/elasticsearch/issues/113325
-- class: org.elasticsearch.integration.KibanaUserRoleIntegTests
-  method: testFieldMappings
-  issue: https://github.com/elastic/elasticsearch/issues/113592
-- class: org.elasticsearch.integration.KibanaUserRoleIntegTests
-  method: testSearchAndMSearch
-  issue: https://github.com/elastic/elasticsearch/issues/113593
 - class: org.elasticsearch.xpack.transform.integration.TransformIT
   method: testStopWaitForCheckpoint
   issue: https://github.com/elastic/elasticsearch/issues/106113

--- a/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/integration/KibanaUserRoleIntegTests.java
+++ b/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/integration/KibanaUserRoleIntegTests.java
@@ -14,13 +14,16 @@ import org.elasticsearch.action.admin.indices.validate.query.ValidateQueryRespon
 import org.elasticsearch.action.search.MultiSearchResponse;
 import org.elasticsearch.cluster.metadata.MappingMetadata;
 import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.core.Strings;
+import org.elasticsearch.index.IndexSettings;
 import org.elasticsearch.index.query.QueryBuilders;
 import org.elasticsearch.test.NativeRealmIntegTestCase;
 import org.elasticsearch.test.SecuritySettingsSourceField;
 import org.elasticsearch.xpack.core.security.authc.support.UsernamePasswordToken;
 
 import java.util.Map;
+import java.util.Random;
 
 import static java.util.Collections.singletonMap;
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertResponse;
@@ -58,6 +61,14 @@ public class KibanaUserRoleIntegTests extends NativeRealmIntegTestCase {
     @Override
     public String configUsersRoles() {
         return super.configUsersRoles() + "my_kibana_user:kibana_user\n" + "kibana_user:kibana_user";
+    }
+
+    @Override
+    protected Settings.Builder setRandomIndexSettings(Random random, Settings.Builder builder) {
+        // Prevent INDEX_CHECK_ON_STARTUP as a random setting since it could result in indices being checked for corruption before opening.
+        // When corruption is detected, it will prevent the shard from being opened. This check is expensive in terms of CPU and memory
+        // usage and causes intermittent CI failures due to timeout.
+        return super.setRandomIndexSettings(random, builder).put(IndexSettings.INDEX_CHECK_ON_STARTUP.getKey(), false);
     }
 
     public void testFieldMappings() throws Exception {

